### ISSUE TITLE
[Bug] Health Will Not Drop to Minimum if Other Mod Increases Health

### ISSUE THREAD

**Issue by BTRBT** [Mrbysco/Unhealthy-Dying#9]
## Client Info
* Minecraft Version:1.12.2

* Forge Version: 14.23.5.2838
* Mod Version: 1.2.2.1
* Java Version: Java 8

## Issue Description
If a player's health is increased with Spice of Life: Carrot Edition, the minimum health value is applied _after_ the increased health. That is, if your minimum health is 2 (ie: One heart) and you gain a heart from SoL, the minimum amount of health that Unhealthy Dying will subsequently reduce you to is 4 (ie: Two hearts). 

## Reproduce Steps
1. Gain an extra heart from Spice of Life: Carrot Edition.
2. Repeatedly commit suicide. Hopefully without too much pain! Oh dear!

## Additional Information
Really love your mod. Keep up the great work, friend-o!

**Comment by Darkmega18** [comment]
Ah, I have this issue also but it seems this is either forgotten or not done.

I have several mods that expand health and I was hoping to have this to reduce health max each time (with a loot table that drops a health increase item that can be stolen by someone else or picked up again later to get the HP back) but unfortunately the HP after it's been subtracted a certain number of times it doesn't continue subtracting health after a time I've noticed in my test world as my health continues to go up for some reason. which was when I realized the bug. :V

**Comment by Mrbysco** [comment]
I am guessing you're using the 1.12 version. Which hasn't gotten the rework.

**Comment by Darkmega18** [comment]
Nope. 1.16.5. :V sorry, didn't realize the report was made from a 1.12.2 version. but mechanically, it is the same bug.

**Comment by Mrbysco** [comment]
When working with other mods that add health it becomes really easy to reduce it and the other mod suddenly taking away it's extra health. which would allow for bugs like a total health of 0 which would probably bug out.

**Comment by Darkmega18** [comment]
I've actually managed to reduce health below 1 previously using scaling health. it just auto sets you to half a heart of health max. Atleast in the case of scaling health. but it had... other issues. some of which became strong compatibility issues and so I switched over to scaling the health of the player and to some degree the mobs a different way.

**Comment by Mrbysco** [comment]
I've recently fixed a bug which allowed negative health. For now the mod will refuse to go lower if a mod has increased your health
